mineinabyss-systems/src/main/kotlin/com/mineinabyss/pvp/survival/SurvivalPvpListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.mineinabyss.pvp.survival
+
+import com.derongan.minecraft.deeperworld.event.PlayerAscendEvent
+import com.derongan.minecraft.deeperworld.event.PlayerDescendEvent
+import com.mineinabyss.components.playerData
+import com.mineinabyss.idofront.messaging.error
+import com.mineinabyss.mineinabyss.core.layer
+import org.bukkit.entity.Player
+import org.bukkit.entity.Projectile
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.entity.EntityDamageByEntityEvent
+
+class SurvivalPvpListener : Listener {
+    @EventHandler
+    fun PlayerDescendEvent.onEnterPvPLayer() {
+        if (toSection.layer?.hasPvPDefault == true && !player.playerData.pvpStatus) {
+            player.error("PVP is always enabled below this point.")
+        }
+    }
+
+    @EventHandler
+    fun PlayerAscendEvent.onEnterOrth() {
+        val data = player.playerData
+        data.showPvPMessage = data.pvpUndecided
+    }
+
+    @EventHandler
+    fun EntityDamageByEntityEvent.playerCombatSystem() {
+        val player = entity as? Player ?: return
+
+        val attacker: Player = when (damager) {
```

Comment:
This should really go into a separate listener that is added by both survival and adventure since the code is the same.